routes/song.js

```diff
@@ -193,13 +201,22 @@ router.route('/:song_id/add-translation')
                         title: data.title,
                         author: data.author,
                         year: data.year,
-                        lang: data.lang,
+                        translator: data.translator,
                         lyric: data.lyric,
+                        lang: data.lang,
+                        youtubeLink: data.youtubeLink,
                         contributor: req.user.username,
                         copyright: data.copyright,
-                        timeAdded: Date.now(),
-                        source: song._id
+                        timeAdded: Date.now()
                     })
+
+                    if (!song.source) {
```

Comment:
Good. I would just add a comment with more explanation here.
We have two cases: when a translation is added to an original versus when a translation is added to a translation of an original. In the first case the source of the original will be null, so we can attach this as a translation of that source. In the second case, the source of the original will point to the true original song and we should make this a translation of that.